proto/index.proto

```diff
@@ -1,23 +1,21 @@
 syntax = "proto3";
 
 import "proto/context.proto";
+import "proto/git.proto";
 
 package index;
 
-message GitRepoRequest {
-  // A string like "buildbuddy-io/buildbuddy".
-  string owner_repo = 1;
-
-  // A commit sha "1d8a3184c996c3d167a281b70a4eeccd5188e5e1".
-  string commit_sha = 2;
+enum ReplacementStrategy {
+  UNKNOWN_REPLACEMENT_STRATEGY = 0;
+  REPLACE_REPO = 1;
+  // TODO(tylerw): support INCREMENTAL
 }
 
-message GitRepoResponse {}
-
 message IndexRequest {
   context.RequestContext request_context = 1;
 
-  GitRepoRequest git_repo = 2;
+  git.GitRepo git_repo = 2;
+  git.RepoState repo_state = 3;
```

Comment:
Removed the weirdly-named bespoke gitRepoRequest and just used the protos we already have for this.